exporters/failure/collector_github.py

```diff
@@ -60,23 +60,23 @@ def __init__(
         else:
             logging.warning("No projects defined for github issues")
             self.projects = []
-        try:
-            user = self._get_github_user()
-        except Exception:
-            logging.warning("github username not found")
-            raise
-        super().__init__(server or self._defaultapi, user, apikey)
 
         self.session = requests.Session()
         # disable .netrc
         self.session.trust_env = False
         if apikey:
             self.session.auth = TokenAuth(apikey)
+        try:
+            user = self._get_github_user()
+        except Exception:
+            logging.warning("github username not found")
+            raise
+        super().__init__(server or self._defaultapi, user, apikey)
 
     def _get_github_user(self) -> str:
         # login and get username
         # set the username / server to env for exporter consistency
-        url = "https://{}/user".format(self.server)
+        url = "https://{}/user".format(self._defaultapi)
```

Comment:
Why do we allow the user to override the github server, only to force them to use `github.com`?